v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/range/BoundarySplitterFactory.java

```diff
@@ -163,13 +163,25 @@ private static Long splitLongs(Long start, Long end) {
   }
 
   private static BigDecimal splitBigDecimal(BigDecimal start, BigDecimal end) {
-    BigInteger startBigInt = (start == null) ? null : start.toBigInteger();
-    BigInteger endBigInt = (end == null) ? null : end.toBigInteger();
-    BigInteger split = splitBigIntegers(startBigInt, endBigInt);
-    if (split == null) {
+    if (start == null && end == null) {
+      return null;
+    }
+    if (start == null) {
+      start = BigDecimal.ZERO;
+    }
+    if (end == null) {
+      end = BigDecimal.ZERO;
+    }
+    int scale = partitionColumn.getScale();
```

Comment:
Are we adding `getScale()` in [PartitionColum](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/96d239d6123580ff52f05d78307a7cbee43b7e40/v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/range/PartitionColumn.java#L45) in some other PR?

If it's not yet added, please add a `scale()` method there and also add a checkState similar to the [one](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/96d239d6123580ff52f05d78307a7cbee43b7e40/v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/range/PartitionColumn.java#L74) that's there